File: leetcode/easy/747.LargestNumberAtLeastTwiceOfOthers/js/largestNumberAtLeastTwiceOfOthers.js
"use strict";
/**
 * @param {number[]} nums
 * @return {number} 
 */

var dominantIndex = function(nums) {
    let greaterValue = -999_999_999;
    let greaterValueIndex = -1;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > greaterValue){
            greaterValueIndex = i;
            greaterValue = nums[i];
        };
    };
    for (let i = 0; i < nums.length; i++){ 
        for (let j = 0; j < nums.length - i - 1; j++){
            if (nums[j] > nums[j + 1]) {
                let temp = nums[j];
                nums[j] = nums[j + 1];
                nums[j + 1] = temp;
            };
        };
    };
    let secondGreaterValue = nums[nums.length - 2];
    if (secondGreaterValue * 2 <= greaterValue) {
        return greaterValueIndex;
    } else {
        return -1;
    };
};

// Testing
let nums = [3,2,1,6];
let nums1 = [1,2,3,4];
let nums2 = [0,0,0,1];
console.log(dominantIndex(nums1));

File: leetcode/easy/747.LargestNumberAtLeastTwiceOfOthers/python/largestNumberAtLeastTwiceOfOthers.py
def dominant_index(nums):
    greater_num = -999
    previous_greater_num = 0
    greater_num_index = -1
    for i in range(len(nums)):
        if nums[i] > greater_num:
            previous_greater_num = greater_num
            greater_num = nums[i]
            greater_num_index = i
        if nums[i] > previous_greater_num and nums[i] < greater_num: 
            previous_greater_num = nums[i]
    if previous_greater_num * 2 <= greater_num:
        return greater_num_index
    else:
        return -1

# Testing
nums = [3,2,1,6]
nums1 = [1,2,3,4]
nums2 = [0,0,0,1]
nums3 = [0,0,3,2]
print(dominant_index(nums3))